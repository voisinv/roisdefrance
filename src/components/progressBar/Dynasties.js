// @flow
import React from 'react';
import * as utils from '../../utils/progressBar';
import { getWindowWidth } from '../../utils/document';
import { getColorPastel } from '../../utils/colors';

export const Dynasty = ({ x, fill, width }: { x: number, fill: string, width: number }) => {
  return (
    <rect
      x={x} y={0}
      width={width}
      height={utils.height}
      fill={fill}>
    </rect>);
};

export default class extends React.PureComponent<any> {

  render() {
    const dynasties = this.props.dynasties;
    const lastDynastie = dynasties[dynasties.length - 1];
    const totalYearsOfReign = lastDynastie.cumulatedYearsOfReign +
      lastDynastie.yearsOfReign;
    const getWidth = utils.getWidth(getWindowWidth(), totalYearsOfReign);
    const getX = utils.getX(getWindowWidth(), totalYearsOfReign);

    return (
      <g>
        {dynasties.map((d, i: number) =>
          <Dynasty
            key={`progress-${d.dynasty}`}
            x={getX(d)}
            width={getWidth(d)}
            fill={getColorPastel(i)}>
          </Dynasty>)}
      </g>);
  }
}