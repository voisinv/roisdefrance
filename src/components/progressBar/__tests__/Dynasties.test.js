import React from 'react';
import renderer from 'react-test-renderer';

import { Dynasty } from '../Dynasties';
import Dynasties from '../Dynasties';

describe('Dynasty', () => {
  it('renders correctly', () => {
    const props = {
      x: 100,
      fill: '#123456',
      width: 200
    };
    const element = renderer.create(<Dynasty {...props}></Dynasty>).toJSON();
    expect(element).toMatchSnapshot();
  });
});

describe('Dynasties', () => {
  it('renders correctly', () => {
    const dynasties = [
      {
        dynasty: 'Merovingiens',
        yearsOfReign: 10,
        cumulatedYearsOfReign: 0
      },
      {
        dynasty: 'Carolingiens',
        yearsOfReign: 5,
        cumulatedYearsOfReign: 10
      }
    ];
    const element = renderer
      .create(<Dynasties dynasties={dynasties}></Dynasties>).toJSON();
    expect(element).toMatchSnapshot();
    expect(element.children.length).toEqual(2);
    expect(element.children[1].type).toEqual('rect');
  });
});