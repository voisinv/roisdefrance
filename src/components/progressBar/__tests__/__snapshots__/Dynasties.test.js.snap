// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dynasties renders correctly 1`] = `
<g>
  <rect
    fill="#b3e2cd"
    height={75}
    width={682.6666666666666}
    x={0}
    y={0}
  />
  <rect
    fill="#fdcdac"
    height={75}
    width={341.3333333333333}
    x={682.6666666666666}
    y={0}
  />
</g>
`;

exports[`Dynasty renders correctly 1`] = `
<rect
  fill="#123456"
  height={75}
  width={200}
  x={100}
  y={0}
/>
`;
